evaluation/evaluate.py

```diff
@@ -194,81 +196,224 @@ def run_evaluation_from_config(working_dir: Path, config: dict, num_questions: i
     # Summary statistics
     with open(results_dir / "summary.json", "w", encoding="utf-8") as summary_file:
         summary_file.write(json.dumps(summary, indent=4))
     logger.info("Evaluation results saved in %s", results_dir)
 
     # Draw the chart for the results
-    fig, ax1 = plt.subplots()
-    ax1.bar(metric_list, pass_rate)
-
-    ax1.set_ylabel("passing rate")
-    ax1.set_title("Passing rate of evaluation metrics")
-    plt.savefig(results_dir / "passing_rate.png")
-    plt.close(fig)
-
-    fig, ax2 = plt.subplots()
-    ax2.bar(metric_list, mean_rate)
-
-    ax2.set_ylabel("mean score")
-    ax2.set_title("Mean score of evaluation metrics")
-    plt.savefig(results_dir / "mean_score.png")
-    plt.close(fig)
-
-    means = {
-        "Max": tuple(max_list),
-        "Min": tuple(min_list),
-        "Mean": tuple(mean_list),
-    }
+    data = [data for _, data in rating_stat_data.items()]
+    titles = [display_metric_name[mn] for mn in rating_stat_data.keys()]
+    y_labels = [metric_value_labels[mn] for mn in rating_stat_data.keys()]
+
+    plot_bar_charts(
+        (1, 3),
+        data,
+        titles,
+        y_labels,
+        [len(questions_with_ratings), 1.0, 5.0],
+        width,
+        results_dir / "evaluation_results.png",
+    )
 
-    x = np.arange(len(metric_name))  # the label locations
-    width = 0.25  # the width of the bars
-    multiplier = 0
-    fig, ax3 = plt.subplots(layout="constrained")
+    gpt_metric_short_names = [short_metric_name[mtrc] for mtrc in gpt_metric_list]
+    plot_radar_chart(
+        gpt_metric_short_names,
+        gpt_metric_mean_list,
+        "GPT Rating Metrics Results",
+        results_dir / "evaluation_gpt_radar.png",
+    )
 
-    for attribute, measurement in means.items():
-        offset = width * multiplier
-        rects = ax3.bar(x + offset, measurement, width, label=attribute)
-        ax3.bar_label(rects, padding=3)
-        multiplier += 1
+    data = [aggregated_data_lists[m] for m in gpt_metric_list]
+    labels = [short_metric_name[m] for m in gpt_metric_list]
+    plot_box_chart(data, "GPT Ratings", labels, "Rating Score", [0.0, 5.0], results_dir / "evaluation_gpt_boxplot.png")
 
-    # Add some text for labels, title and custom x-axis tick labels, etc.
-    ax3.set_title("Evaluation results")
-    ax3.set_xticks(x + width, tuple(metric_name))
-    ax3.legend(loc="upper left", ncols=3)
-    ax3.set_ylim(0, 250)
+    data = [data for _, data in aggregated_data_lists.items()]
+    titles = [display_metric_name[mn] for mn in aggregated_data_lists.keys()]
+    y_labels = [metric_value_labels[mn] for mn in aggregated_data_lists.keys()]
 
-    plt.savefig(results_dir / "eval.png")
-    plt.close(fig)
+    plot_box_charts(
+        (1, 3),
+        data[len(gpt_metric_list) :],
+        titles[len(gpt_metric_list) :],
+        y_labels[len(gpt_metric_list) :],
+        results_dir / "evaluation_stat_boxplot.png",
+    )
```

Comment:
Bug: the layout is fixed to (1,3), if we only select other numbers of numeric metrics, it will cause an Error. Mark here, will take a look at how to fix this later